metaflow/runtime.py

```diff
@@ -238,8 +245,32 @@ def execute(self):
                                         'by the end of flow.')
 
     def _killall(self):
+        # If we are here, all children have received a signal and are shutting down.
+        # We want to give them an opportunity to do so and then kill
+        live_workers = []
+        temp_workers = {}
         for worker in self._workers.values():
-            worker.kill()
+            temp_workers[worker] = 1
+        live_workers = temp_workers.keys()
```

Comment:
isn't `live_workers = self._workers.values()`?

Reply:
Not quite; in workers, they appear twice (one for each FD). This dedups them.

Reply:
so, `live_workers = set(self._workers.values())`

Reply:
Could you also detail what `temp_workers` is for? If it's for the sole purpose of creating the set, I'd instead do what @savingoyal suggests w.r.t set.